runtime/shared_common/OSCache.cpp

```diff
@@ -1197,3 +1225,104 @@ void *
 SH_OSCache::getOSCacheStart() {
 	return _headerStart;
 }
+
+/**
+ * Get the unique ID of the current cache
+ * @param[in] currentThread  The current VM thread.
+ *
+ * @return the cache unique ID
+ */
+const char*
+SH_OSCache::getCacheUniqueID(J9VMThread* currentThread)
+{
+	PORT_ACCESS_FROM_VMC(currentThread);
+	if (NULL != _cacheUniqueID) {
+		return _cacheUniqueID;
+	}
+	Trc_SHR_Assert_True(NULL != _cacheDirName);
+	Trc_SHR_Assert_True(NULL != _cacheName);
+
+	U_32 cacheType = J9_ARE_ALL_BITS_SET(_runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_PERSISTENT_CACHE) ? J9PORT_SHR_CACHE_TYPE_PERSISTENT : J9PORT_SHR_CACHE_TYPE_NONPERSISTENT;
+	UDATA sizeRequired = generateCacheUniqueID(currentThread, _cacheDirName, _cacheName, _layer, cacheType, NULL, 0);
+
+	_cacheUniqueID = (char*)j9mem_allocate_memory(sizeRequired, J9MEM_CATEGORY_VM);
+	if (NULL == _cacheUniqueID) {
+		return NULL;
+	}
+	generateCacheUniqueID(currentThread, _cacheDirName, _cacheName, _layer, cacheType, _cacheUniqueID, sizeRequired);
+	return _cacheUniqueID;
+}
+
+/**
+ * Generate a cache unique ID
+ * @param[in] currentThread  The current VM thread.
+ * @param[in] cacheDirName  The cache directory
+ * @param[in] cacheName  The cache name
+ * @param[in] layer  The cache layer number
+ * @param[in] cacheType  The cache type
+ * @param[out] buf  The buffer for the cache unique ID
+ * @param[out] bufLen  The length of the buffer
+ *
+ * @return If buf is not NULL, the number of characters printed into buf is returned , not including the NUL terminator.
+ * 			If buf is NULL, the size of the buffer required to print to the unique ID, including the NUL terminator is returned.
+ */
+UDATA
+SH_OSCache::generateCacheUniqueID(J9VMThread* currentThread, const char* cacheDirName, const char* cacheName, I_8 layer, U_32 cacheType, char* buf, UDATA bufLen)
+{
+	char nameWithVGen[J9SH_MAXPATH];
+	char cacheFilePathName[J9SH_MAXPATH];
+	J9JavaVM* vm = currentThread->javaVM;
+	PORT_ACCESS_FROM_VMC(currentThread);
+
+	J9PortShcVersion versionData;
+	setCurrentCacheVersion(vm, J2SE_VERSION(vm), &versionData);
+	versionData.cacheType = cacheType;
+
+	getCacheVersionAndGen(PORTLIB, vm, nameWithVGen, J9SH_MAXPATH, cacheName, &versionData, OSCACHE_CURRENT_CACHE_GEN, true, layer);
+	/* Directory is included here, so if the cache directory is renamed, caches with layer > 0 becomes unusable */
+	getCachePathName(PORTLIB, cacheDirName, cacheFilePathName, J9SH_MAXPATH, nameWithVGen);
+	I_64 timeStamp = j9file_lastmod((const char*)cacheFilePathName);
+	I_64 fileSize = j9file_length((const char*)cacheFilePathName);
```

Comment:
The casts to `(const char *)` should be redundant; please remove them.

Reply:
Fixed